os/kernel/log_dump/log_dump.c

```diff
@@ -169,21 +200,88 @@ int log_dump_save(char ch)
 			current_ptr = 0; /* reset the read pointer as head is modified */
 			node->arr[0] = ch;
 			sq_addlast((sq_entry_t *)node, &log_dump_chunks);
+			if (set_comp_head) { /* new node, so compressed block starts at 0 */
+				node->comp_head = 0;
+				set_comp_head = false;
+			} else {
+				node->comp_head = -1;
+			}
 			log_dump_mem_check(max_size);
 		}
 
 	} else { /* there is place in tail */
-
+		struct log_dump_chunk_s *log_dump_tail = (struct log_dump_chunk_s *)sq_tail(&log_dump_chunks);
```

Comment:
Seems to be extra spaces at the start of the line. Could you please check?

Reply:
done